dts/bindings/gpio/altr,pio-1.0.yaml

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: Altera PIO node
+
+compatible: "altr,pio-1.0"
+
+include: [gpio-controller.yaml, base.yaml]
+
+properties:
+  reg:
+    required: true
+
+  direction:
+    type: string
+    description: |
+      Direction can be set to Bidir, Input, InOut or Output.
+      It cannot be configure when during the runtime.
+      This is to make sure the driver is match with the hardware.
+    default: output
+    enum:
+      - "bidir"
+      - "input"
+      - "inout"
+      - "output"
```

Comment:
I take it by "hardware" you mean the synthesized FPGA IP core? If yes, I think it's worth mentioning (as hardware could also be the peripherals connected to the FPGA pads).

Reply:
i have changed the description as below:
  direction:
    type: string
    description: |
      Direction can be set to bidir, input, inout or output.
      It cannot be configure when during the runtime and should be
      set according to the synthesized FPGA design.